src/ctapipe/calib/camera/extractor.py

```diff
@@ -0,0 +1,228 @@
+
+        # median over the sample per pixel
+        pixel_median = np.ma.median(masked_images, axis=0)
+
+        # mean over the sample per pixel
+        pixel_mean = np.ma.mean(masked_images, axis=0)
+
+        # std over the sample per pixel
+        pixel_std = np.ma.std(masked_images, axis=0)
+
+        # outliers from median
+        image_median_outliers = np.logical_or(
+            pixel_median < self.image_median_cut_outliers[0],
+            pixel_median > self.image_median_cut_outliers[1],
+        )
+
+        return StatisticsContainer(
+            extraction_start=times[0],
+            extraction_stop=times[-1],
+            mean=pixel_mean.filled(np.nan),
+            median=pixel_median.filled(np.nan),
+            median_outliers=image_median_outliers.filled(True),
+            std=pixel_std.filled(np.nan),
+        )
+
+
+class SigmaClippingExtractor(StatisticsExtractor):
+    """
+    Extract the statistics from a chunk of images
+    using astropy's sigma clipping functions
+    """
+
+    max_sigma = Int(
+        default_value=4,
+        help="Maximal value for the sigma clipping outlier removal",
+    ).tag(config=True)
+    iterations = Int(
+        default_value=5,
+        help="Number of iterations for the sigma clipping outlier removal",
+    ).tag(config=True)
+
+    def _extract(self, images, times, masked_pixels_of_sample) -> StatisticsContainer:
+        # ensure numpy array
+        masked_images = np.ma.array(images, mask=masked_pixels_of_sample)
+
+        # mean, median, and std over the sample per pixel
+        pixel_mean, pixel_median, pixel_std = sigma_clipped_stats(
+            masked_images,
+            sigma=self.max_sigma,
+            maxiters=self.iterations,
+            cenfunc="mean",
+            axis=0,
+        )
+
+        # mask pixels without defined statistical values
+        pixel_mean = np.ma.array(pixel_mean, mask=np.isnan(pixel_mean))
+        pixel_median = np.ma.array(pixel_median, mask=np.isnan(pixel_median))
+        pixel_std = np.ma.array(pixel_std, mask=np.isnan(pixel_std))
+
+        # median of the median over the camera
+        median_of_pixel_median = np.ma.median(pixel_median, axis=1)
+
+        # median of the std over the camera
+        median_of_pixel_std = np.ma.median(pixel_std, axis=1)
+
+        # std of the std over camera
+        std_of_pixel_std = np.ma.std(pixel_std, axis=1)
+
+        # outliers from median
+        image_deviation = pixel_median - median_of_pixel_median[:, np.newaxis]
+        image_median_outliers = np.logical_or(
+            image_deviation
+            < self.image_median_cut_outliers[0]  # pylint: disable=unsubscriptable-object
+            * median_of_pixel_median[:, np.newaxis],
+            image_deviation
+            > self.image_median_cut_outliers[1]  # pylint: disable=unsubscriptable-object
+            * median_of_pixel_median[:, np.newaxis],
+        )
```

Comment:
After some offline discussions with @FrancaCassol, we noticed that here we need to distinguish between checking for outliers for signal (flatfield) and noise (pedestal) using `median_of_pixel_median` and `std_of_pixel_std`, respectively.

Reply:
Resolved by [e8497fd](https://github.com/cta-observatory/ctapipe/pull/2554/commits/e8497fde9386d17644bdf8bc02deba736658cdf3).